presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/WindowMatcher.java

```diff
@@ -24,20 +26,83 @@
 
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 
 import static java.util.Objects.requireNonNull;
 
 final class WindowMatcher
         implements Matcher
 {
+    private final WindowNode.Specification specification;
     private final List<FunctionCall> functionCalls;
 
-    WindowMatcher(List<FunctionCall> functionCalls)
+    WindowMatcher(WindowNode.Specification specification, List<FunctionCall> functionCalls)
     {
+        this.specification = specification;
         this.functionCalls = ImmutableList.copyOf(requireNonNull(functionCalls, "functionCalls is null"));
     }
 
+    private static boolean sameIshSpecification(
```

Comment:
Same partition and order, but yeah. This is what you get for reviewing rough code ;-)